controllers/controller_test.go

```diff
@@ -199,11 +199,13 @@ var _ = Describe("snr Controller", func() {
 				node.Status.Conditions[0].Reason = "foo"
 				Expect(k8sClient.Client.Status().Update(context.Background(), node)).To(Succeed())
 
-				By("Verify that finalizer was removed and SNR can be deleted")
-				testNoFinalizer()
+				removeUnschedulableTaint(node)
```

Comment:
Hmm I'm not clear about that.
I think it should be done in the actual code (not test code), otherwise `verifyNoExecuteTaintRemoved` doesn't make much sense

Reply:
this is the NoSchedule taint, simulating the node lifecycle controller, just like with adding it a few line above 🙂